raphtory/src/db/api/view/graph.rs

```diff
@@ -135,76 +143,211 @@ impl<'graph, G: BoxableGraphView + Sized + Clone + 'graph> GraphViewOps<'graph>
     }
 
     fn materialize(&self) -> Result<MaterializedGraph, GraphError> {
-        let g = GraphStorage::default();
-        let earliest = if let Some(earliest) = self.earliest_time() {
-            earliest
+        let storage = self.core_graph().lock();
+        let g = TemporalGraph::default();
+        if let Some(earliest) = self.earliest_time() {
+            g.update_time(TimeIndexEntry::start(earliest));
         } else {
-            return Ok(self.new_base_graph(g));
+            return Ok(self.new_base_graph(g.into()));
         };
 
-        // make sure we preserve all layers even if they are empty
-        // skip default layer
-        for layer in self.unique_layers().skip(1) {
-            g.resolve_layer(Some(&layer))?;
-        }
-        // Add edges first so we definitely have all associated nodes (important in case of persistent edges)
-        for e in self.edges() {
-            // FIXME: this needs to be verified
-            for ee in e.explode_layers() {
-                let layer_id = *ee.edge.layer().expect("exploded layers");
-                let layer_ids = LayerIds::One(layer_id);
-                let layer_name = self.get_layer_name(layer_id);
-                let layer_name: Option<&str> = if layer_id == 0 {
-                    None
-                } else {
-                    Some(&layer_name)
-                };
-
-                for ee in ee.explode() {
-                    g.add_edge(
-                        ee.time().expect("exploded edge"),
-                        ee.src().id(),
-                        ee.dst().id(),
-                        ee.properties().temporal().collect_properties(),
-                        layer_name,
-                    )?;
-                }
+        if let Some(latest) = self.latest_time() {
+            g.update_time(TimeIndexEntry::end(latest));
+        } else {
+            return Ok(self.new_base_graph(g.into()));
+        };
```

Comment:
need to move this after copying the graph constant properties

Reply:
and add a test

Reply:
also need to copy the graph temporal properties here (and add more tests)

Reply:
check that the next_event_id returns something sensible after materialise

Reply:
Exactly what I was thinking!